app/services/courses/generate_course_title_service.rb

```diff
@@ -0,0 +1,80 @@
+module Courses
+  class GenerateCourseTitleService
+    def execute(course:)
+      subjects = course.subjects
+
+      title = if course.further_education_course?
+                further_education_title
+              elsif is_modern_lanuage_course?(subjects)
+                modern_language_title(subjects)
+              else
+                generated_title(subjects)
+              end
+
+      title = append_send_info(title) if course.is_send?
+      title
+    end
+
+  private
+
+    def is_modern_lanuage_course?(subjects)
+      subjects.any? { |s| s == SecondarySubject.modern_languages }
+    end
+
+    def modern_language_title(subjects)
+      title = "Modern Languages"
+
+      languages = subjects.select { |s| s.type == "ModernLanguagesSubject" }
+
+      return title if languages.empty? || languages.length >= 4
+
+      languages = languages.map { |language| format_language_name(language) }
```

Comment:
I prefer to avoid modifying a variable if the meaning changes, `languages` -> `language_names`